### PR TITLE
Add `return-in-computed-property` rule.

### PR DESCRIPTION
**Pull request by armano2**
This PR implements rules proposed in #75

**DONE:**
- [x] Basic check for `return`
- [x] Check if all paths `return` value
- [x] Option: `treatUndefinedAsUnspecified` - disallows implicitly returning undefined with a `return;` statement.

### CODE REVIEW

**Comment by armano2** [comment]
@michalsnik can you review this pull request?

**Comment by michalsnik** [comment]
Sure, later today I'll try to push my PR and both of yours forward. Busy time :)

**Comment by armano2** [comment]
It's my first time working with rules from eslint (developing them) but they seems not to complex to make.

sorry for pushing you to test my code 🗡

File: docs/rules/computed-property-return.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Enforces that a return statement is present in computed property (computed-property-return)
```
Comment (michalsnik) [inline docs/rules/computed-property-return.md]: I'd call this rule `return-in-computed-property` to make it more descriptive. WDYT?

File: docs/rules/computed-property-return.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+```
+
+## :wrench: Options
+
+This rule has an object option:
+- `"treatUndefinedAsUnspecified"`: `true` (default) allows implicitly returning undefined with a `return;` statement.
+
+```
+vue/order-in-components: [2, {
```
Comment (michalsnik) [inline docs/rules/computed-property-return.md]: Whoops

File: docs/rules/computed-property-return.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      return false
+    }
+  }
+}
+```
+
+## :wrench: Options
+
+This rule has an object option:
+- `"treatUndefinedAsUnspecified"`: `true` (default) allows implicitly returning undefined with a `return;` statement.
```
Comment (michalsnik) [inline docs/rules/computed-property-return.md]: I think we should force to return something explicitly, thus in my opinion `return;` should also trigger an error and this option should not be necessary.
We could consider two types of messages:
- There is no `return` statement inside `foo` computed property
- There is an incomplete `return` statement inside `foo` computed property
 What do you think @armano2 @chrisvfritz ?

Comment (armano2) [inline docs/rules/computed-property-return.md]: damn description is invalid, i forgot to change it, by default its treating undefined as unspecified than `return;` will thrown an error, but i prefer to leave a choice to disable it because undefined can be useful in some cases.

File: lib/rules/computed-property-return.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        if (!isValidReturn() && funcInfo.codePath.currentSegments.some((segment) => segment.reachable)) {
+          forbiddenNodes.push({
+            hasReturn: funcInfo.hasReturn,
+            node: funcInfo.node,
+            type: 'return'
+          })
+        }
+      }
+    },
+    utils.executeOnVueInstance(context, properties => {
```
Comment (michalsnik) [inline lib/rules/computed-property-return.js]: Shouldn't it be `executeOnVueComponent` ? And this one in `computed-data` rule instead?

File: docs/rules/computed-property-return.md (hunk 1 above)
Comment (armano2) [inline docs/rules/computed-property-return.md]: i'm bad at naming stuff, your name is way better

File: lib/rules/computed-property-return.js (hunk 4 above)
Comment (armano2) [inline lib/rules/computed-property-return.js]: no right now there is 2 functions, `executeOnVueInstance` and `executeOnVueComponent` we have to distinct both of those types, 

see: https://github.com/vuejs/eslint-plugin-vue/pull/78/commits/281f230ef92b3d034c28c7014f11da66269682d1

Comment (michalsnik) [inline lib/rules/computed-property-return.js]: Yeah, but I thought this matters for `data` as function case, as it's ok to have data as object in vue instance, but not in component. And this rule is about `return` statement in computed property which seems irrelevant.

Comment (armano2) [inline lib/rules/computed-property-return.js]: `executeOnVueInstance` supports `components` & `new vue`

i just renamed function to simplified merges and improve readability.

**Comment by armano2** [comment]
@michalsnik cleaned up and merged

**Comment by chrisvfritz** [comment]
Will this also flag computed properties that technically have a return somewhere in their content, but not directly? For example:

``` js
computed: {
  foo: function () {
    function bar () {
      return this.baz * 2 
    }
    bar()
  }
}
```

**Comment by armano2** [comment]
@chrisvfritz yes (test for this case added)

File: tests/lib/rules/return-in-computed-property.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            }
+          }
+        }
+      `,
+      parserOptions: { ecmaVersion: 8, sourceType: 'module' },
+      errors: [{
+        message: 'Expected to return a value in "foo" computed property.',
+        line: 4
+      }]
+    }
```
Comment (mysticatea) [inline tests/lib/rules/return-in-computed-property.js]: Could you add invalid tests for `treatUndefinedAsUnspecified` option?

Comment (armano2) [inline tests/lib/rules/return-in-computed-property.js]: ok

Comment (armano2) [inline tests/lib/rules/return-in-computed-property.js]: @mysticatea added
